main.cpp

```diff
@@ -360,6 +309,44 @@ FloatType& FloatType::divide(float x)
     return *this;
 }
 
+//double
+DoubleType::DoubleType(double v) : value(new double(v)) {}
+
+DoubleType::~DoubleType() 
+{
+    delete value;
+}
+
+DoubleType& DoubleType::pow(double y)
+{
+    powInternal(y);
+    return *this;
+}
+
+DoubleType& DoubleType::pow(const FloatType& y)
+{
+    powInternal(static_cast<double>(y));
+    return *this;
+}
+
+DoubleType& DoubleType::pow(const DoubleType& y)
+{
+    powInternal(static_cast<double>(y));
+    return *this;
+}
+
+DoubleType& DoubleType::pow(const IntType& y)
+{
+    powInternal(static_cast<double>(y));
+    return *this;
+}
```

Comment:
`powInternal` returns `*this` already. 
you don't need a separate return statement.
```
return powInternal(...);
```
- [ ] revise